01_Provisioning/src/module/api.ts

```diff
@@ -69,71 +72,54 @@ export class ProvisioningModuleApi extends AbstractModuleApi<ProvisioningModule>
         while (setVariableData.length > 0) {
-            const setVariableDataSubset = setVariableData.slice(0, itemsPerMessageSetVariables);
-            const messageConfirmation = await this._module.sendCall(identifier, tenantId, CallAction.SetVariables,
-                { setVariableData: setVariableDataSubset } as SetVariablesRequest);
-            if (!messageConfirmation.payload) {
-                finalMessageConfirmation.success = false;
-                finalMessageConfirmation.payload = finalMessageConfirmation.payload ?
-                    (finalMessageConfirmation.payload as SetVariableDataType[]).concat(setVariableDataSubset) : setVariableDataSubset
-            }
+            await this._module.sendCall(identifier, tenantId, CallAction.SetVariables,
+                { setVariableData: setVariableData.slice(0, itemsPerMessageSetVariables) } as SetVariablesRequest, callbackUrl);
+            setVariableData = setVariableData.slice(itemsPerMessageSetVariables);
         }
-        return finalMessageConfirmation;
+        return { success: true };
     }
 
     @AsMessageEndpoint(CallAction.GetVariables, GetVariablesRequestSchema)
     async getVariables(
         identifier: string,
         tenantId: string,
-        request: GetVariablesRequest
+        request: GetVariablesRequest,
+        callbackUrl?: string
     ): Promise<IMessageConfirmation> {
-        const getVariableData = request.getVariableData;
+        let getVariableData = request.getVariableData as GetVariableDataType[];
         let itemsPerMessageGetVariables = await this._module._deviceModelService.getItemsPerMessageGetVariablesByStationId(identifier);
 
         // If ItemsPerMessageGetVariables not set, send all variables at once
         itemsPerMessageGetVariables = itemsPerMessageGetVariables == null ?
             getVariableData.length : itemsPerMessageGetVariables;
 
-        // We don't wait for each message to be responded to, instead this is fire-and-forget.
-        // Only a failure from the message bus to the central system is caught, downstream failures are invisible.
-        const finalMessageConfirmation: IMessageConfirmation = { success: true };
         while (getVariableData.length > 0) {
-            const getVariableDataSubset = getVariableData.slice(0, itemsPerMessageGetVariables);
-            const messageConfirmation = await this._module.sendCall(identifier, tenantId, CallAction.GetVariables,
-                { getVariableData: getVariableDataSubset } as GetVariablesRequest);
-            if (!messageConfirmation.payload) {
-                finalMessageConfirmation.success = false;
-                finalMessageConfirmation.payload = finalMessageConfirmation.payload ?
-                    (finalMessageConfirmation.payload as GetVariableDataType[]).concat(getVariableDataSubset) : getVariableDataSubset
-            }
+            await this._module.sendCall(identifier, tenantId, CallAction.GetVariables,
```

Comment:
@thanaParis what if this `sendCall` fails? We seem to only return `success` here?